Source/WebCore/Modules/webtransport/WebTransport.cpp

```diff
@@ -156,24 +163,27 @@ void WebTransport::receiveDatagram(std::span<const uint8_t> datagram)
     m_datagramSource->receiveDatagram(datagram);
 }
 
-void WebTransport::receiveIncomingUnidirectionalStream(Ref<ReadableStreamSource>&& incomingStream)
+void WebTransport::receiveIncomingUnidirectionalStream(WebTransportStreamIdentifier identifier)
 {
     auto* globalObject = scriptExecutionContext()->globalObject();
     if (!globalObject)
         return;
     auto& jsDOMGlobalObject = *JSC::jsCast<JSDOMGlobalObject*>(globalObject);
+    auto incomingStream = WebTransportReceiveStreamSource::create();
     auto stream = [&] {
         Locker<JSC::JSLock> locker(jsDOMGlobalObject.vm().apiLock());
-        return WebTransportReceiveStream::create(jsDOMGlobalObject, WTFMove(incomingStream));
+        return WebTransportReceiveStream::create(jsDOMGlobalObject, incomingStream.copyRef());
     } ();
     if (stream.hasException())
         return;
     auto receiveStream = stream.releaseReturnValue();
     m_receiveStreams.add(receiveStream);
-    m_receiveStreamSource->receiveIncomingStream(jsDOMGlobalObject, WTFMove(receiveStream));
+    ASSERT(!m_readStreamSources.contains(identifier));
+    m_readStreamSources.set(identifier, WTFMove(incomingStream));
```

Comment:
```suggestion
    m_readStreamSources.add(identifier, WTFMove(incomingStream));
```
Given the map cannot contain identifier?